src/event.rs

```diff
@@ -67,6 +67,16 @@ pub enum Event<'a, T: 'static> {
         event: DeviceEvent,
     },
 
+    /// Emitted on macOS to let the application know that a file or files are requested to be opened.
```

Comment:
I think this first line should be generic and not platform specific. Mentioning that this is only supported on macOS should go in a platform-specific notice at the bottom. At least that's the way things are usually documented in winit.